extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/WindowOperatorQueryFrameProcessor.java

```diff
@@ -213,7 +213,8 @@ Current approach with R&C and operators materialize a single R&C for processing.
      Most of the window operations like SUM(), RANK(), RANGE() etc. can be made with 2 passes of the data. We might think to reimplement them in the MSQ way so that we do not have to materialize so much data.
      */
 
-    if (partitionColumnNames.isEmpty()) {
+    // todo: need to cleanup unused code and comments
+    if (true) {
```

Comment:
@kgyrtkirk Which layer should the `maxRowsMaterialized` check be enforced now?

With the current PR changes, most tests in `WindowOperatorQueryFrameProcessorTest` fail because we are passing the entire MSQ partition to the list of operators.

Do we need to move `maxRowsMaterialized` check to the `GlueingPartitioningOperator` layer? If yes, we'd need to fetch the `maxRowsMaterialized` from query context and use it while creating `GlueingPartitioningOperator` in the Windowing.java layer?

Thoughts?

Reply:
Based on offline discussion, maxRowsMaterialized check should be moved to the glueing operator.

But on further thoughts, the same can't be done in Windowing.java as we don't have access to `MultiStageQueryContext.MAX_ROWS_MATERIALIZED_IN_WINDOW` and `Limits.MAX_ROWS_MATERIALIZED_IN_WINDOW`

Another potential option which was coined was changing `OperatorFactory#wrap` interface method signature from `Operator wrap(Operator op)` to `Operator wrap(Operator op, Map<String, Object> context)`, and Windowing can just add a PartitioningFactory whose `wrap` method makes the decision between Naive and Glueing partitioning operators. Will have to look into this, if we have all the data to make this approach work or not.

Reply:
Based on further offline discussion, the plan for now is to replace NaivePartitioningOperatorFactory with GlueingPartitioningOperatorFactory in WindowOperatorQueryKit layer, as that layer has access to everything (maxRowsMaterialized and corresponding MSQ constants). Will work on making this change.